contracts/psm/VenusAdapter.sol

```diff
@@ -0,0 +1,180 @@
+    address public token; // token address
+    address public vToken; // vToken address
+    uint256 public quota; // quota
+
+    uint256 public quotaAmount; // quota amount
+
+    bytes32 public constant MANAGER = keccak256("MANAGER"); // manager role
+
+    event Deposit(uint256 amount);
+    event Withdraw(address account, uint256 amount);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    modifier onlyVaultManager() {
+        require(msg.sender == vaultManager, "only VaultManager can call this function");
+        _;
+    }
+
+    /**
+     * @dev initialize contract
+     * @param _admin admin address
+     * @param _manager manager address
+     * @param _vaultManager vault manager address
+     * @param _venusPool venus pool address
+     * @param _token token address
+     * @param _vToken vToken address
+     */
+    function initialize(
+            address _admin,
+            address _manager,
+            address _vaultManager, 
+            address _venusPool, 
+            address _token, 
+            address _vToken,
+            uint256 _quotaAmount
+        ) public initializer {
+        require(_admin != address(0), "admin cannot be zero address");
+        require(_manager != address(0), "manager cannot be zero address");
+        require(_vaultManager != address(0), "vaultManager cannot be zero address");
+        require(_venusPool != address(0), "venusPool cannot be zero address");
+        require(_token != address(0), "token cannot be zero address");
+        require(_vToken != address(0), "vToken cannot be zero address");
+
+        __AccessControl_init();
+        __UUPSUpgradeable_init();
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+        _setupRole(MANAGER, _manager);
+
+        vaultManager = _vaultManager;
+        token = _token;
+        venusPool = _venusPool;
+        vToken = _vToken;
+        quotaAmount = _quotaAmount;
+    }
+
+    /**
+     * @dev deposit token by vault manager
+     * @param amount deposit amount
+     */
+    function deposit(uint256 amount) external onlyVaultManager {
+        require(amount > 0, "deposit amount cannot be zero");
+        IERC20(token).safeTransferFrom(vaultManager, address(this), amount);
+        IERC20(token).safeIncreaseAllowance(venusPool, amount);
+
+        // deposit to venus pool
+        IVBep20Delegate(venusPool).mint(amount);
+
+        emit Deposit(amount);
+    }
+
+    /**
+     * @dev withdraw token by vault manager
+     * @param account withdraw account
+     * @param amount withdraw amount
+     */
+    function withdraw(address account, uint256 amount) external onlyVaultManager {
+        require(amount > 0, "withdraw amount cannot be zero");
+
+        uint256 exchangeRate = IVBep20Delegate(venusPool).exchangeRateStored();
+        // calculate vToken amount
+        uint256 vTokenAmount = Math.mulDiv(amount, 1e18, exchangeRate);
```

Comment:
is it necessary to "+1" in this calculation ?

Reply:
When venus exchange rate is converted, the precision lost will be greater than 1 wei.
So we should use quota to ensure withdraw amount == real transfer amount.